packages/cli/src/telemetry/index.ts

```diff
@@ -0,0 +1,126 @@
+/// <reference lib="dom" />
+import fs from 'fs-extra';
+import uuid from 'uuid/v4';
+import path from 'path';
+import crypto from 'crypto';
+import envPaths from 'env-paths';
+import mapValues from 'lodash.mapvalues';
+import inquirer from 'inquirer';
+import proc from 'child_process';
+
+import { DISABLE_INTERACTIVITY } from '../prompts/prompt';
+import { Params } from '../scripts/interfaces';
+import ProjectFile from '../models/files/ProjectFile';
+
+type Field = string | number | boolean;
+
+interface GlobalTelemetryOptions {
+  optIn: boolean;
+  uuid: string;
+  salt: string;
+}
+
+export type CommandData = Params & {
+  name: string;
+  network?: string;
+};
+
+export default {
+  async report(commandName: string, options: Params, interactive: boolean): Promise<void> {
+    const telemetry = await checkOptIn(interactive);
+    if (telemetry === undefined || !telemetry.optIn) return;
```

Comment:
This being called `telemetry` can be confusing, since in other contexts that name is the whole module, and functions like `telemetry.report` are called. Perhaps we could rename this to `telemetryOptions`?